cpp/include/cuvs/neighbors/cagra.hpp

```diff
@@ -104,6 +105,16 @@ struct index_params : cuvs::neighbors::index_params {
                graph_build_params::ivf_pq_params,
                graph_build_params::nn_descent_params>
     graph_build_params;
+
+  /**
+   * Whether to add the dataset content to the index, i.e.:
+   *
+   *  - `true` means the index is filled with the dataset vectors and ready to search after calling
+   * `build`.
+   *  - `false` means `build` only builds the graph, but
+   * the user is expected to update dataset separately.
+   */
+  bool add_data_on_build = true;
```

Comment:
I think we need a different parameter name here. `add_data_on_build` is [already defined for the index_params](https://github.com/rapidsai/cuvs/blob/581c9cc9ca23c1d828d28f1ad84e1c53fe5077c6/cpp/include/cuvs/neighbors/common.hpp#L101), and I believe it has a slightly different meaning. It's meant to be used together with `extend` function; in IVF-* methods we allow adding any new data after clustering this way. In CAGRA in the current form, the user would only be allowed to add the data that is present in the graph (and not via `extend` function).

Or, at the very least, use explicit `using cuvs::neighbors::index_params::add_data_on_build;` instead of shadowing the parent definition.

Reply:
I have moved `add_data_on_build` to the ivf `index_params` and added a new arg for cagra -- `populate_data`. The PR is also green on CI right now.